### PR TITLE
Update spellcasting entries to change proficiency ranks

### PR DESCRIPTION
**Pull request by stwlam**
Also move display of spellcasting rank in proficiencies tab to own subsection

### CODE REVIEW

File: src/module/actor/character/sheet.ts [hunk 1]
```diff
@@ -754,6 +754,35 @@ class CharacterSheetPF2e<TActor extends CharacterPF2e> extends CreatureSheetPF2e
             }
         }
 
+        // Update all "normal" spellcasting entries
+        for (const select of htmlQueryAll<HTMLSelectElement>(html, "select[data-action=update-spellcasting-rank]")) {
```
Comment (CarlosFdez) [inline src/module/actor/character/sheet.ts:758]:
```suggestion
        for (const select of htmlQueryAll<HTMLSelectElement>(html, "select[data-action=change-spellcasting-rank]")) {
```

File: src/module/actor/character/sheet.ts [hunk 2]
```diff
@@ -754,6 +754,35 @@ class CharacterSheetPF2e<TActor extends CharacterPF2e> extends CreatureSheetPF2e
...
+                        "Item",
+                        entries.map((e) => ({ _id: e.id, "system.proficiency.value": newRank })),
+                    );
+                }
+            });
+        }
+
+        // Update a single spellcasting entry that uses an alternative statistic
+        for (const select of htmlQueryAll<HTMLSelectElement>(
+            castingPanel,
```
Comment (CarlosFdez) [inline src/module/actor/character/sheet.ts:773]: why castingPanel instead of html like the other?

Comment (stwlam) [reply src/module/actor/character/sheet.ts:773]: These are only in the spellcasting panel

Comment (CarlosFdez) [reply src/module/actor/character/sheet.ts:773]: then should the other one also refer to castingPanel?

File: static/templates/actors/character/tabs/spellcasting.hbs [hunk 3]
```diff
@@ -66,7 +66,11 @@
                         <section class="spell-ability">
                             <span class="spell-tradition">{{localize entry.statistic.label}}</span>
                             <div class="spellcasting-prof button-group">
-                                <select class="skill-proficiency pf-rank adjust-item-stat-select" data-item-property="system.proficiency.value" data-rank="{{entry.statistic.rank}}" data-dtype="Number">
+                                <select
+                                    class="skill-proficiency pf-rank"
+                                    data-action="{{#if entry.usesSpellProficiency}}update-spellcasting-rank{{else}}update-spellcasting-entry{{/if}}"
```
Comment (CarlosFdez) [inline static/templates/actors/character/tabs/spellcasting.hbs:71]: I think you can avoid a specific handler for the regular update if you use data-item-property

**Pull request merged**